File: backend/src/main/java/com/eventflow/modules/checkin/api/dto/CheckInDtos.java
package com.eventflow.modules.checkin.api.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;

import java.time.Instant;

/** DTOs espejo de CheckInRequest/CheckInResponse (OpenAPI congelado). */
public final class CheckInDtos {

    private CheckInDtos() {
    }

    public record CheckInRequest(@NotBlank String qrToken) {
    }

    @JsonInclude(JsonInclude.Include.NON_NULL)
    public record CheckInResponse(String result, String attendeeName, String ticketTypeName,
                                  String zoneName, String denialCode, Instant occurredAt) {
    }
}

File: app/src/main/java/com/app/eventflow/data/remote/dto/checkin/CheckInDtos.kt
package com.app.eventflow.data.remote.dto.checkin

import kotlinx.serialization.Serializable

/** DTOs espejo de QrResponse/CheckInRequest/CheckInResponse (OpenAPI congelado). */

@Serializable
data class QrResponseDto(val qrToken: String, val expiresAt: String, val refreshAfter: String)

@Serializable
data class CheckInRequestDto(val qrToken: String)

@Serializable
data class CheckInResponseDto(
    val result: String,
    val attendeeName: String? = null,
    val ticketTypeName: String? = null,
    val zoneName: String? = null,
    val denialCode: String? = null,
    val occurredAt: String,
)
